modules/core/02-client/migrations/v7/store.go

```diff
@@ -0,0 +1,219 @@
+		return clienttypes.ErrClientNotFound
+	}
+
+	var clientState exported.ClientState
+	if err := cdc.UnmarshalInterface(bz, &clientState); err != nil {
+		return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into tendermint client state")
+	}
+
+	_, ok := clientState.(*ibctm.ClientState)
+	if !ok {
+		return sdkerrors.Wrap(clienttypes.ErrInvalidClient, "client state is not tendermint even though client id contains 07-tendermint")
+	}
+
+	return nil
+}
+
+// handleLocalhostMigration removes all client and consensus states associated with the localhost client type.
+func handleLocalhostMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.BinaryCodec) error {
+	clients, err := collectClients(ctx, store, Localhost)
+	if err != nil {
+		return err
+	}
+
+	for _, clientID := range clients {
+		clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
+		clientStore := prefix.NewStore(store, clientPrefix)
+
+		// delete the client state
+		clientStore.Delete(host.ClientStateKey())
+
+		pruneClientConsensusStates(clientStore)
+	}
+
+	return nil
+}
+
+// collectClients will iterate over the provided client type prefix in the client store
+// and return a list of clientIDs associated with the client type. This is necessary to
+// avoid state corruption as modifying state during iteration is unsafe. A special case
+// for tendermint clients is included as only one tendermint clientID is required for
+// this migration..
+func collectClients(ctx sdk.Context, store sdk.KVStore, clientType string) (clients []string, err error) {
+	clientPrefix := []byte(fmt.Sprintf("%s/%s", host.KeyClientStorePrefix, clientType))
+	iterator := sdk.KVStorePrefixIterator(store, clientPrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		path := string(iterator.Key())
+		if !strings.Contains(path, host.KeyClientState) {
+			// skip non client state keys
+			continue
+		}
+
+		clientID, err := host.ParseClientStatePath(path)
+		if err != nil {
+			return nil, err
+		}
+
+		clients = append(clients, clientID)
+
+		// optimization: exist after a single tendermint client iteration
+		if strings.Contains(clientID, exported.Tendermint) {
+			return clients, nil
+		}
+	}
+
+	return clients, nil
+}
+
+// pruneClientConsensusStates removes all client consensus states from the associated
+// client store which is keyed by the clientID.
+func pruneClientConsensusStates(clientStore sdk.KVStore) {
+	iterator := sdk.KVStorePrefixIterator(clientStore, []byte(host.KeyConsensusStatePrefix))
+	var heights []exported.Height
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+		// key is in the format "consensusStates/<height>"
+		if len(keySplit) != 2 || keySplit[0] != string(host.KeyConsensusStatePrefix) {
+			continue
+		}
+
+		// collect consensus states to be pruned
+		heights = append(heights, clienttypes.MustParseHeight(keySplit[1]))
```

Comment:
existing code from `legacy/v100/store.go`, it can be cleaned up in a different pr (matches iteration in keeper functions)